src/NSubstitute/Core/Maybe.cs

```diff
@@ -27,7 +28,8 @@ public Maybe(T value) : this()
         public Maybe<T> OrElse(Maybe<T> other) { return OrElse(() => other); }
         public T ValueOr(Func<T> other) { return Fold(other, x => x); }
         public T ValueOr(T other) { return ValueOr(() => other); }
-        public T ValueOrDefault() { return ValueOr(default(T)); }
+        [return: MaybeNull]
+        public T ValueOrDefault() { return ValueOr(default(T)!); }
```

Comment:
**question:** can this be `T?`

Reply:
The exclamation mark operator is called [null-forgiving operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-forgiving). It does not assert that type is not null, it just tell a compiler to "forgive" a particular violation. It might be confusing at the first glance, as I would also expect `!` be perceived as "not-null assertion operator".

In this particular case complier does not allow `null` for the `T` type. Therefore our `ValueOr()` method does not allow passing `null` as an argument. We use the null-forgiving operator here to suppress that check here, as we are fine to return "default".